### PR TITLE
Feature/onboarding templates

### PR DESCRIPTION
**Pull request by shogunpurple**
## Description
This PR contains code that introduces an onboarding modal that allows you to use templates pulled from https://github.com/Budibase/templates. Adding new templates in here does not require redeploying budibase, as it pulls directly from the manifest in that repository. So we can add lots of new templates in there directly.

I've added the IT Help Desk and Applicant tracking templates to that repo.

Right now this modal will show when you have no apps in your budibase portal, but before we merge this we should decide whether we want to show this (not the initial welcome screen) as part of the app creation modal, or to only show it **once**, and when the user clicks out of it, we mark them as onboarded in the DB, and don't show them it again. Either one is trivial to update.

## TODO

- [x] Add all relevant templates to https://github.com/Budibase/templates for launch

- [x] Decide on when exactly to show this modal to the user. IMO we should make this the create app modal, and also show it to them when they have no apps - possibly even replacing the "You have no apps" notice in the center of the screen on the portal.
- [x] Update the images for each template. I'm just using a placeholder for now, but @joebudi can give me the real ones. I'm not sure how to get them from the XD dev mode

## Notes
- The template a user chooses will be sent up in our analytics, so we know which templates are the most popular.

## Screenshots
Screenshot 2021-10-05 at 22 46 55

Screenshot 2021-10-05 at 23 15 36

### CODE REVIEW

File: packages/builder/src/components/start/OnboardingModal.svelte [hunk 1]
```diff
@@ -0,0 +1,71 @@
+<script>
+  import { ModalContent, Body } from "@budibase/bbui"
+  // import { auth } from "stores/portal"
```
[the remark this answers is not in the document]
Comment (shogunpurple) [reply packages/builder/src/components/start/OnboardingModal.svelte:3]: Yeah this is really just for either uncommenting or deletion based on the logic that we go for to show this modal

File: packages/builder/src/components/start/OnboardingModal.svelte [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  <ModalContent
+    title={`Welcome ${$auth.user?.firstName + "!" || "!"}`}
+    confirmText="Next"
+    onConfirm={nextStep}
+  >
+    <Body size="S">
+      <p>Welcome to Budibase!</p>
+      <p>
+        We're different from other development tools in some really special
+        ways, so we'd like to take you through them.
```
[the remark this answers is not in the document]
Comment (Rory-Powell) [reply packages/builder/src/components/start/OnboardingModal.svelte:41]: ah nice, I wasn't aware we were using intercom for that!

[the remark this answers is not in the document]
Comment (Rory-Powell) [reply packages/builder/src/components/start/OnboardingModal.svelte:41]: is there somewhere we can log in to see where that stuff lives?

File: packages/builder/src/components/start/OnboardingModal.svelte (hunk 1 above)
[the remark this answers is not in the document]
Comment (Rory-Powell) [reply packages/builder/src/components/start/OnboardingModal.svelte:3]: I think it's needed regardless, see the github actions warning for
```js
  title={`Welcome ${$auth.user?.firstName + "!" || "!"}`}
```

**Comment by aptkingston** [comment]
> 
> 
> > Decide on when exactly to show this modal to the user. IMO we should make this the create app modal, and also show it to them when they have no apps
> 
> Agree on this. I'm used to this sort of experience when using apps like Miro, Canva, etc. If we were to do the second part about showing when there is no apps I would prefer it to be embedded in the main panel rather than being a modal.

I agree aout not forcing a popup modal when it's empty, I think having it inline in that case would be much nicer. The `Modal` component already has an `inline` prop to make it behave like this.

**Review by aptkingston — APPROVED** [review]
Great stuff Marty. I agree we should make templates always part of the create app modal, at least for now. Down the line if we split the apps section into Joe's idea of drafts / apps / templates and therefore had a dedicated section for templates then it would make sense to remove it from the create app modal, but I like including it there for now.

I also like the idea of basically making it the empty state screen, and like I said in the comment above we could try using an inline modal for that, rather than agressively popping it up whenever you navigate to that screen. Worth a shot anyway to see how it feels.

File: packages/builder/src/components/start/OnboardingModal.svelte (hunk 2 above)
[the remark this answers is not in the document]
Comment (shogunpurple) [reply packages/builder/src/components/start/OnboardingModal.svelte:34]: It doesn't exist yet - Joe is still working on it. But I might be able to create you an account on intercom if you want to see it.. Not sure if we are charged by seat or not

**Comment by shogunpurple** [comment]
Cheers @aptkingston - I've moved the modal inline to be the default flow if you have no apps, which should encourage people to create them. 

Screenshot 2021-10-06 at 11 05 53
👍

Will add the template option to the create app modal as well.

**Comment by shogunpurple** [comment]
After some input from @joebudi this has been redesigned to look a bit cleaner. Removed the inital welcome screen and used a list with icons for templates instead:

image

**Pull request merged**